sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -279,99 +321,432 @@ public EventData addContext(String key, Object value) {
+     * considered active and therefore not retrievable until the scheduled enqueue time. Mind that the message may not
+     * be activated (enqueued) at the exact given datetime; the actual activation time depends on the queue's workload
+     * and its state.
+     * </p>
+     *
+     * @return the datetime at which the message will be enqueued in Azure Service Bus
+     */
+    public OffsetDateTime getScheduledEnqueueTime() {
+        Object value = amqpAnnotatedMessage.getMessageAnnotations().get(SCHEDULED_ENQUEUE_UTC_TIME_NAME.getValue());
+        return value != null
+            ? ((OffsetDateTime) value).toInstant().atOffset(ZoneOffset.UTC)
+            : null;
+    }
+
+    /**
+     * Sets the scheduled enqueue time of this message. A {@code null} will not be set. If this value needs to be unset
+     * it could be done by value removing from {@link AmqpAnnotatedMessage#getMessageAnnotations()} using key {@link
+     * AmqpMessageConstant#SCHEDULED_ENQUEUE_UTC_TIME_NAME}.
+     *
+     * @param scheduledEnqueueTime the datetime at which this message should be enqueued in Azure Service Bus.
+     *
+     * @return The updated {@link EventData}.
+     * @see #getScheduledEnqueueTime()
+     */
+    public EventData setScheduledEnqueueTime(OffsetDateTime scheduledEnqueueTime) {
+        if (scheduledEnqueueTime != null) {
+            amqpAnnotatedMessage.getMessageAnnotations().put(SCHEDULED_ENQUEUE_UTC_TIME_NAME.getValue(),
+                scheduledEnqueueTime);
         }
+        return this;
+    }
+
+    /**
+     * Sets a partition key for sending a message to a partitioned entity
+     *
+     * @param partitionKey The partition key of this message.
+     *
+     * @return The updated {@link EventData}.
+     * @throws IllegalArgumentException if {@code partitionKey} is too long or if the {@code partitionKey} does not
+     *     match the {@code sessionId}.
+     * @see #getPartitionKey()
+     */
+    public EventData setPartitionKey(String partitionKey) {
+        checkIdLength("partitionKey", partitionKey, MAX_PARTITION_KEY_LENGTH);
+        checkPartitionKey(partitionKey);
+
+        amqpAnnotatedMessage.getMessageAnnotations().put(PARTITION_KEY_ANNOTATION_NAME.getValue(), partitionKey);
+        return this;
+    }
 
-        /**
-         * Gets the offset within the Event Hubs stream.
-         *
-         * @return The offset within the Event Hubs stream.
-         */
-        private Long getOffset() {
-            return offset;
+    /**
+     * Gets or sets a session identifier augmenting the {@link #getReplyTo() ReplyTo} address.
+     * <p>
+     * This value augments the {@link #getReplyTo() reply to} information and specifies which {@code sessionId} should
+     * be set for the reply when sent to the reply entity.
+     *
+     * @return The {@code getReplyToGroupId} property value of this message.
+     */
+    public String getReplyToSessionId() {
```

Comment:
EH has no this thing

Reply:
According to your comment, remove unnecessary parameters, variables and methods in the new version